src/robothub_oak/application.py

```diff
@@ -31,90 +39,120 @@ def on_start(self) -> None:
                 # Make sure it is properly closed in case it disconnected during runtime
-                self.__close_oak(device_mxid)
+                self.close_device(device_mxid)
 
+                self.on_device_connected(device_mxid)
                 # Connect to the device
                 self.__connect(device_mxid)
 
+                self.on_device_disconnected(device_mxid)
+
                 # If device is connected
                 if self.__devices[device_mxid]:
-                    log.debug(f'Device {device_mxid}: creating pipeline...')
+                    log.debug(f'Device {product_name}: creating pipeline...')
 
                     self.setup_pipeline(device=self.__devices[device_mxid])
                     self.__devices[device_mxid].start(blocking=False)
-                    log.info(f'Device {device_mxid}: started successfully.')
+                    log.info(f'Device {product_name}: started successfully.')
 
                     # Threads for polling and reporting
                     polling_thread = Thread(target=self.__poll_device,
                                             args=(self.__devices[device_mxid],),
                                             daemon=True,
                                             name=f'poll_{device_mxid}')
                     polling_thread.start()
-                    self.__polling_threads.append(polling_thread)
 
                     reporting_thread = Thread(target=self.__device_stats_reporting,
                                               args=(device_mxid,),
                                               daemon=True,
                                               name=f'stats_reporting_{device_mxid}')
                     reporting_thread.start()
-                    self.__reporting_threads.append(reporting_thread)
                 else:
                     self.wait(25)
 
             self.wait(5)
 
-        self.__close_oak(mxid=device_mxid)
-        log.debug(f'Device {device_mxid}: thread stopped.')
+        self.close_device(mxid=device_mxid)
+        log.debug(f'Device {product_name}: thread stopped.')
 
-    def __poll_device(self, device: OakCamera):
+    def __poll_device(self, device: OakCamera) -> None:
+        """
+        Poll the device for new data. This method is called in a separate thread.
+
+        :param device: The device to poll.
+        """
         while self.running:
             if not device.poll():
                 return
 
             time.sleep(0.0025)
 
-    def __device_stats_reporting(self, device_mxid: str):
+    def __device_stats_reporting(self, device_mxid: str) -> None:
+        """
+        Report device info and stats every 30 seconds.
+
+        :param device_mxid: The mxid of the device to report info/stats for.
+        """
         dai_device = self.__devices[device_mxid].device
         state = self.__device_states[device_mxid]
         while self.running:
```

Comment:
Reporting thread doesn't exit when device disconnects. It will just throw, catch and log exceptions.